absl/container/internal/container_memory.h

```diff
@@ -438,29 +438,29 @@ struct map_slot_policy {
     // but std::pair is not trivially copyable in C++23 in some standard
     // library versions.
     // See https://github.com/llvm/llvm-project/pull/95444 for instance.
-    auto is_relocatable = typename std::conjunction<
+    static constexpr auto kIsRelocatable = typename std::conjunction<
         absl::is_trivially_relocatable<typename value_type::first_type>,
         absl::is_trivially_relocatable<typename value_type::second_type>>::
         type();
 
     emplace(new_slot);
-    if (is_relocatable) {
+    if constexpr (kIsRelocatable) {
```

Comment:
This is now producing a warning:
```
./absl/container/internal/container_memory.h:434:35: error: parameter 'alloc' set but not used [-Werror=unused-but-set-parameter]
  434 |   static auto transfer(Allocator* alloc, slot_type* new_slot,
```

It seems that whatever analysis GCC is doing doesn't understand branches.

I'm guessing ``[maybe_unused]`` will fix this. Or perhaps `kIsRelocatable` could be a `bool` and the `std::conjunction` should use `::value` instead of `::type()`, maybe that would also give GCC a hint?